stores/current.ts

```diff
@@ -159,29 +159,51 @@ function round2(value: number): number {
 }
 
 function round4(value: number): number {
+  if (typeof value !== 'number') {
+    throw new TypeError(`Expected a number but received ${typeof value}`);
+  }
+
   return parseFloat(value.toFixed(4))
 }
 
-function roundValue<V extends Partial<LchValue>>(
-  value: V,
-  type: 'lch' | 'oklch'
-): V {
-  let rounded = { ...value }
-  if (typeof rounded.l !== 'undefined') {
-    rounded.l = round2(rounded.l)
-  }
-  if (typeof rounded.c !== 'undefined') {
-    rounded.c = type === 'oklch' ? round4(rounded.c) : round2(rounded.c)
-  }
-  if (typeof rounded.h !== 'undefined') {
-    rounded.h = round2(rounded.h)
-  }
-  if (typeof rounded.a !== 'undefined') {
-    rounded.a = round2(rounded.a)
+
+function incrementalRoundValue(value: LchValue, COLOR_FN: string): LchValue {
+  const maxIterations = 6;
+
+  for (let i = 0; i <= maxIterations; i++) {
+    let precision = i === 0 ? 0 : Math.min(i, maxIterations);
+    let roundedValue: LchValue = {
+      a: round2(value.a),
+      c: round2(value.c),
```

Comment:
I am not sure, that we should use the same round on each L, C, H on every step.

The problem is that `L` is `0-100`, while `C` is `0-1`. So it better to use different round level for `C` on every step. 

Otherwise, the `incrementalRoundValue` will keep too many digits after `,` for `L`, because algorithm can’t round `C` anymore.